### PR TITLE
Reorganize `connect` subcommand for readibility

### PR DESCRIPTION
**Pull request by yaahc**

### CODE REVIEW

**Comment by hdevalence** [comment]
Reposting some comments from discord:

Since the network interface is much more mature, it could be a good starting point.  I think all we really need is a way to use Abscissa's global state management to pass around handles to whatever services are provided by each component crate.  This could be done by making wrapper components and adding a method to each one that would hand out a boxed `Service`, so you could do, e.g.,

```
app_reader()
    .state()
    .components
    .get_downcast_ref::<NetworkComponent>()
    .expect("network should be available")
    .service(); // returns Box<dyn Service<....> + Clone>
```

Cloning the services and handing out clones avoids a problem where using a `Service` requires mutable access, but the only way to get mutable access is through a global `app_writer` lock.

File: zebrad/src/commands/connect.rs [hunk 1]
```diff
@@ -98,32 +98,31 @@ impl ConnectCmd {
 
 type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
 
-struct Connect<ZN, ZNF, ZS> {
+struct Connect<ZN, ZS>
+where
+    ZN: Service<zebra_network::Request>,
```
Comment (hdevalence) [inline zebrad/src/commands/connect.rs:103]: any reason to have just a partial bound here rather than the full bound or no bound?

Comment (hdevalence) [reply zebrad/src/commands/connect.rs:103]: ah, `ZN: Service` so that `ZN::Future` is available below.

File: zebrad/src/commands/connect.rs [hunk 2]
```diff
@@ -98,32 +98,31 @@ impl ConnectCmd {
...
-            Error = Error,
-            Future = ZNF,
-        > + Send
+    ZN: Service<zebra_network::Request, Response = zebra_network::Response, Error = Error>
+        + Send
         + Clone
         + 'static,
-    ZNF: Future<Output = Result<zebra_network::Response, Error>> + Send,
-    ZS: Service<zebra_state::Request, Response = zebra_state::Response, Error = Error, Future = ZSF>
+    ZN::Future: Future<Output = Result<zebra_network::Response, Error>> + Send,
```
Comment (hdevalence) [inline zebrad/src/commands/connect.rs]:
```suggestion
    ZN::Future: Send,
```
does this bound work? the `Future<...>` is implied by `Service` i think

File: zebrad/src/commands/connect.rs [hunk 3]
```diff
@@ -98,32 +98,31 @@ impl ConnectCmd {
...
+    ZS: Service<zebra_state::Request, Response = zebra_state::Response, Error = Error>
         + Send
         + Clone
         + 'static,
-    ZSF: Future<Output = Result<zebra_state::Response, Error>> + Send,
+    ZS::Future: Future<Output = Result<zebra_state::Response, Error>> + Send,
```
Comment (hdevalence) [inline zebrad/src/commands/connect.rs]:
```suggestion
    ZS::Future: Send,
```

**Review by hdevalence — DISMISSED** [review]
This seems fine to me.  However, because it's a throwaway stub command, I think it would be preferable not to try to clean up the `connect` code too much, though, and instead try to get the `start` command implemented.  At some point we may want to stick other code in place of the current connect stub, at which point having the `Connect` struct doesn't really help much.

File: zebrad/src/commands/connect.rs [hunk 4]
```diff
@@ -24,7 +34,6 @@ impl Runnable for ConnectCmd {
     fn run(&self) {
         info!(connect.addr = ?self.addr);
 
-        use crate::components::tokio::TokioComponent;
```
Comment (dconnolly) [inline zebrad/src/commands/connect.rs:27]: 👍👍

**Review by dconnolly — COMMENTED** [review]
Good pending the trimming of bounds per @hdevalence's suggested changes.

File: zebrad/src/commands/connect.rs (hunk 2 above)
Comment (yaahc) [reply zebrad/src/commands/connect.rs]: oo, fancy

**Review by dconnolly — APPROVED** [review]
🚤

**Pull request merged**